pallets/disputes/src/lib.rs

```diff
@@ -0,0 +1,466 @@
+            })?;
+
+            Disputes::<T>::mutate(dispute_key, |maybe_dispute| {
+                if let Some(d) = maybe_dispute {
+                    d.expiration = new_expiry;
+                    d.is_extended = true
+                }
+            });
+
+            Self::deposit_event(Event::<T>::DisputeExtended { dispute_key });
+            Ok(())
+        }
+    }
+
+    #[derive(Encode, Decode, PartialEq, Eq, Clone, Debug, TypeInfo, MaxEncodedLen)]
+    #[scale_info(skip_type_params(T))]
+    pub struct Dispute<T: Config> {
+        /// Who this was raised by.
+        pub raised_by: AccountIdOf<T>,
+        /// The votes of each jury.
+        pub votes: BoundedVotes<T>,
+        /// The party responsible for the vote.
+        pub jury: BoundedVec<AccountIdOf<T>, <T as Config>::MaxJurySize>,
+        /// The specific entities the dispute is raised upon.
+        pub specifiers: BoundedVec<T::SpecificId, T::MaxSpecifics>,
+        /// Marks wether a dispute has already had its time limit extended.
+        pub is_extended: bool,
+        /// The expiration block of the dispute.
+        pub expiration: BlockNumberFor<T>,
+    }
+
+    impl<T: Config> Dispute<T> {
+        // Create a new dispute and setup state so that pallet will operate as intended.
+        pub(crate) fn new(
+            dispute_key: T::DisputeKey,
+            raised_by: AccountIdOf<T>,
+            jury: BoundedVec<AccountIdOf<T>, T::MaxJurySize>,
+            specifiers: BoundedVec<T::SpecificId, T::MaxSpecifics>,
+        ) -> Result<(), DispatchError> {
+            ensure!(
+                !Disputes::<T>::contains_key(dispute_key),
+                Error::<T>::DisputeAlreadyExists
+            );
+
+            let expiration_block =
+                frame_system::Pallet::<T>::block_number().saturating_add(T::VotingTimeLimit::get());
+            let dispute = Self {
+                raised_by,
+                jury,
+                votes: Default::default(),
+                specifiers,
+                is_extended: false,
+                expiration: expiration_block,
+            };
+
+            Disputes::<T>::insert(dispute_key, dispute);
+            DisputesFinaliseOn::<T>::try_mutate(expiration_block, |b_vec| {
+                b_vec
+                    .try_push(dispute_key)
+                    .map_err(|_| Error::<T>::TooManyDisputesThisBlock)?;
+
+                Ok::<(), DispatchError>(())
+            })?;
+
+            crate::Pallet::<T>::deposit_event(Event::<T>::DisputeRaised { dispute_key });
+            Ok(())
+        }
+
+        /// Calculate the winner in a dispute at the current moment.
+        /// This ofcourse is subject to change if more votes are had.
+        pub(crate) fn calculate_winner(&self) -> DisputeResult {
+            if self.votes.values().len() == 0 {
+                return DisputeResult::Failure;
+            }
+
+            if self.votes.values().filter(|&&x| x).count()
```

Comment:
what happens if only 1 member in the dispute?

Reply:
also instead of X, can we use `is_yay` or something similar?

Reply:
@mshankarrao

Reply:
done